tools/linter/adapters/constexpr_linter.py

```diff
@@ -0,0 +1,96 @@
+"""
+CONSTEXPR: Ensures users don't use vanilla constexpr since it causes issues
+"""
+
+import argparse
+import json
+import logging
+from os import replace
+import sys
+import re
+
+from enum import Enum
+from typing import NamedTuple, Optional
+
+CONSTEXPR = "constexpr"
```

Comment:
should we restrict it to `constexpr char` as constexprs for ints and floats are fine?

Reply:
Updated